source/code/plugins/baseline_lib.rb

```diff
@@ -0,0 +1,117 @@
+require 'json'
+require 'securerandom' # SecureRandom.uuid 
+
+require_relative 'oms_common'
+
+module OMS
+    class Baseline
+
+        @@log = nil
+        def self.log= (value)
+	        @@log = value
+        end
+
+        # ------------------------------------------------------
+        def self.transform_and_wrap(results, host, time)		
+            if results == nil
+                @@log.debug "Baseline Assessment failed; Empty input"
+                return nil, nil
+            end
+
+            if results["results"] == nil 
+                @@log.debug "Baseline Assessment failed; Invalid input:" + results.inspect
+                return nil, nil
+            end
+
+            results["assessment_id"] = SecureRandom.uuid      
+    
+            asm_baseline_results = results["results"]
+            scan_time = results["scan_time"]
+            assessment_id = results["assessment_id"]
+
+            baseline_blob = {
+                "DataType"=>"SECURITY_BASELINE_BLOB", 
+                "IPName"=>"Security",
+                "DataItems"=>[
+                ]
+            }	
+
+            asm_baseline_results.each do |asm_baseline_result|
+                oms_baseline_result = self.transform_asm_2_oms(asm_baseline_result, scan_time, host, assessment_id)		
+                baseline_blob["DataItems"].push(oms_baseline_result)
+            end 
+
+            baseline_summary_blob = self.calculate_summary(results, host, time)
+
+            @@log.debug "Baseline Summary: " + baseline_summary_blob.inspect	
+
+            return baseline_blob, baseline_summary_blob
+        end # self.transform_and_wrap
+    
+        # ------------------------------------------------------	   
+        def self.calculate_summary(results, hostname, time)  

```

Comment:
the time parameter is not used anywhere in the method. shouldn't this be added to the summary somehow?
As TimeGenerated maybe?